### PR TITLE
Use common staged manager for provisioning process

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- use common staged manager for provisioning process
- change all provisioning steps interface to fulfill common manager requiremenets

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/avs/delegator.go [hunk 1]
```diff
@@ -97,9 +97,6 @@ func (del *Delegator) AddTags(log logrus.FieldLogger, operation internal.Provisi
 			return op, duration, err
 		}
 	}
-
-	updatedOperation, d = del.provisionManager.SimpleUpdateOperation(operation)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/avs/delegator.go:101]: this method does not updates operation, so no need to do a storage call

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

File: components/kyma-environment-broker/internal/metrics/step_result.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func (c *StepResultCollector) OnOperationStepProcessed(ctx context.Context, ev i
 	}
 
 	switch {
-	case stepProcessed.Operation.Type == "provisioning":
+	case stepProcessed.Operation.Type == internal.OperationTypeProvision:
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/metrics/step_result.go:124]: potentially culprit why `compass_keb_provisioning_result` metrics disappeared recently. The old switch statement was comparing to `"provisioning"` but `internal.OperationTypeProvision` is defined as `provision` in
https://github.com/kyma-project/control-plane/blob/b40b0c560b1b1986acdd7ed657306a4b33982204/components/kyma-environment-broker/internal/model.go#L168

**Review by wozniakjan — COMMENTED** [review]
https://github.tools.sap/kyma/backlog/issues/2939

File: components/kyma-environment-broker/internal/metrics/step_result.go (hunk 2 above)
Comment (wozniakjan) [reply components/kyma-environment-broker/internal/metrics/step_result.go:124]: likely similar to deprovisioning

Comment (ebensom) [reply components/kyma-environment-broker/internal/metrics/step_result.go:124]: Nice catch!

**Pull request merged**